beacon_node/beacon_chain/src/beacon_chain.rs

```diff
@@ -3702,8 +3760,58 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
         match availability {
             Availability::Available(block) => {
                 publish_fn()?;
-                // Block is fully available, import into fork choice
-                self.import_available_block(block).await
+                let block_root = block.block_root();
+                match self.block_status_table.transition(
+                    &block_root,
+                    BlockStatus::Pending,
+                    BlockStatus::Importing,
+                ) {
+                    Ok(_) => {
+                        let result = self.import_available_block(block).await;
+                        if let Ok(AvailabilityProcessingStatus::Imported(_)) = result.as_ref() {
```

Comment:
This is equivalent to a `match` with a default statement, which have been very problematic in the past as we modify the variants. Can you do a full match on result?